validate_email/mx_check.py

```diff
@@ -119,70 +122,87 @@ def _smtp_converse(
             raise _ProtocolError('connect', code, message)
         _smtp_ehlo_tls(smtp=smtp, helo_host=helo_host)
         _smtp_mail(smtp=smtp, from_address=from_address)
-        code, message = smtp.rcpt(recip=email_address.ace)
-        if code == 250:
-            # Address valid, early exit
-            raise StopIteration
-        elif code >= 500:
-            raise _ProtocolError('RCPT TO', code, message)
-
-
-def _check_one_mx(
```

Comment:
The reason why I had this function is to keep the functions smaller than ~24 lines per function. If you end up with a huge function, it is much harder to understand how it works. Can you revert this with your added changes?

Reply:
Splitting this into two functions would mean that I have to pass a lot of things back and forth, for example I'd have to pass both error collection variables into _check_one_mx so they can be filled there. I first tried it that way, but I had the impression that the code is actually harder to understand with split functions. However, if you think it's better, I can of course change it.

Reply:
Sounds like a candidate for an entire class with methods for me then ... If it gets this complicated, classes provide a better way for communication between the smaller parts.

Reply:
Interesting idea, I think I'll give it a try. (Not today, though)

Reply:
Take your time :) I'm not Mr. Know It All, these are just my suggestions according to my experiences :) I always try to keep it clean and simple, so that anybody who comes into the project can easily contribute.